retrieval/k8s/k8s_stats.go

```diff
@@ -124,6 +126,38 @@ func writeK8sResourceFile(workDir *os.File, resourceName string,
 	return err
 }
 
+//nolint:gocyclo
+func shouldSkipResource(k8Resource interface{}) bool {
+	// safe buffer to allow for longer lived resources to be ingested correctly
+	previousHour := time.Now().Add(-1 * time.Hour)
+	switch resource := k8Resource.(type) {
+	case *v1batch.Job:
+		if resource.Status.CompletionTime != nil &&
+			previousHour.After(resource.Status.CompletionTime.Time) {
+			return true
+		}
+		if resource.Status.Failed > 0 && resource.Status.StartTime != nil &&
+			previousHour.After(resource.Status.StartTime.Time) {
+			return true
+		}
```

Comment:
what if this is a long running job and it just failed in this hour? And timestamp for failure time?

Reply:
If it has failed during the current sample period then we wouldn't record it in any case, as the pods will have already stopped. I'm not opposed to bumping it back out further in any case though if we think additional headroom could be valuable.

Reply:
It is not about the `headroom`. I just feel the `startTime` here doesn't make sense.
If your point is we don't record failed pod/job usage, we could ignore it directly. (though I thought we use it and have roadmap to identify crash pods/jobs which consume the resources)